crates/wiggle/src/lib.rs

```diff
@@ -522,17 +533,19 @@ impl<'a, T> GuestPtr<'a, [T]> {
     /// This function will return a `GuestSlice` into host memory if all checks
     /// succeed (valid utf-8, valid pointers, memory is not borrowed, etc). If
     /// any checks fail then `GuestError` will be returned.
-    pub fn as_slice(&self) -> Result<GuestSlice<'a, T>, GuestError>
+    pub fn as_slice(&self) -> Result<Option<GuestSlice<'a, T>>, GuestError>
     where
         T: GuestTypeTransparent<'a>,
     {
-        let len = match self.pointer.1.checked_mul(T::guest_size()) {
-            Some(l) => l,
-            None => return Err(GuestError::PtrOverflow),
-        };
-        let ptr =
-            self.mem
-                .validate_size_align(self.pointer.0, T::guest_align(), len)? as *mut T;
+        if self.mem.is_shared_memory() {
+            return Ok(None);
+        }
```

Comment:
Could this have a comment, or a reference, for why shared memory is rejected here?